contentScript.js

```diff
@@ -58,14 +63,19 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     switch (request.message) {
         case 'refresh_data':
             // the locationchange event causes the manager integration to rerun
-            window.dispatchEvent(new Event('locationchange'));
+            window.postMessage({
+                source: 'evoTools',
+                type: 'evolv:refreshContext'
+            }, '*');
             break;
         case 'enable_evolv':
+            window.sessionStorage.removeItem('evolv:blockExecution');
             chrome.storage.sync.remove('evolv:blockExecution');
             window.location.reload();
             break;
         case 'disable_evolv':
-            chrome.storage.sync.set({'evolv:blockExecution': true});
+            window.sessionStorage.setItem('evolv:blockExecution', 'true');
+            chrome.storage.sync.set({ "evolv:blockExecution": true });
```

Comment:
seems like you should only need to synchronize the value and it would pull this form session storage?

Reply:
contentscript.js window object is the client webpage but popup.js window object is popup window object. So the shared storage is `chrome.storage.sync.set({ "evolv:blockExecution": true });`